tests/include/test/threading_helpers.h

```diff
@@ -0,0 +1,104 @@
+/**
+ * \file threading_helpers.h
+ *
+ * \brief This file contains the prototypes of helper functions for the purpose
+ *        of testing threading.
+ */
+
+/*
+ *  Copyright The Mbed TLS Contributors
+ *  SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+ */
+
+#ifndef THREADING_HELPERS_H
+#define THREADING_HELPERS_H
+
+#if defined MBEDTLS_THREADING_C
+
+#include "mbedtls/private_access.h"
+#include "mbedtls/build_info.h"
+
+/* Most fields of publicly available structs are private and are wrapped with
+ * MBEDTLS_PRIVATE macro. This define allows tests to access the private fields
+ * directly (without using the MBEDTLS_PRIVATE wrapper). */
+#define MBEDTLS_ALLOW_PRIVATE_ACCESS
+
+#define MBEDTLS_ERR_THREADING_THREAD_ERROR                 -0x001F
+
+#if defined(MBEDTLS_THREADING_PTHREAD)
+#include <pthread.h>
+
+typedef struct mbedtls_test_thread_t {
+    pthread_t MBEDTLS_PRIVATE(thread);
+} mbedtls_test_thread_t;
+
+#endif /* MBEDTLS_THREADING_PTHREAD */
```

Comment:
Minor improvement; you can ignore it if you don't agree.
With this design the definition of `mbedtls_test_thread_t` is to be duplicated for every threading library that is going to be supported. What about creating this structure as:

```
typedef struct mbedtls_test_thread_t {
#if defined(MBEDTLS_THREADING_PTHREAD)
    pthread_t MBEDTLS_PRIVATE(thread);
#elif defined(<some-other-threading-lib>)
    [...]
#else
    unsigned dummy;
#endif /* MBEDTLS_THREADING_PTHREAD */
} mbedtls_test_thread_t;
```

The `dummy` case is for when `MBEDTLS_THREADING_C` is defined, but no threading solution is (no `MBEDTLS_THREADING_PTHREAD` or any others). That line might be avoided if this condition is not allowed in `check_config.h`.